src/sparseml/core/recipe/recipe.py

```diff
@@ -580,3 +651,48 @@ def _parse_recipe_from_md(file_path, yaml_str):
             " {}".format(file_path)
         )
     return yaml_str
+
+
+def create_recipe_string_from_modifiers(
+    modifiers: List[Modifier],
+    modifier_group_name: Optional[str] = None,
+) -> str:
+    """
+    Create a recipe string from a list of Modifier instances
+
+    (Note: this pathway assumes there's only one stage in the recipe
+    associated by the modifier_group_name, if None, a dummy default
+    group_name will be assigned.)
+
+    :param modifiers: The list of Modifier instances
+    :param modifier_group_name: The stage_name of the recipe,
+        if `oneshot` or `train` the run_type of the recipe will be
+        inferred from the modifier_group_name, if None, a dummy default
+        group_name will be assigned.
+    :return: A string in yaml format from which the recipe can be created
+    """
+
+    # Recipe(s) are yaml/json strings of the following format:
+    # run_type_stage: # should contain oneshot/train
+    #    modifiers:
+    #        ModifierTypeOne:
+    #            start: 0.0
+    #            end: 2.0
+    #            ...
+    #        ModifierTypeTwo:
+    #            ...
+
+    # Create a recipe string from the modifiers
+    # Default greoup is sparseml_intermediate
+    default_group_name: str = "sparseml_intermediate"
+    modifier_group_name: str = modifier_group_name or default_group_name
+
+    recipe_dict = {
+        f"{modifier_group_name}_stage": {
+            "pruning_modifiers": {
```

Comment:
could we make these defaults more generic? ie default_stage and default_modifiers

Reply:
Also, do we need to be able to support staged modifiers at all @bfineran? I could see that working by being able to pass in a dictionary of list(Modifiers) keyed on stage names

Reply:
For now, that's not in scope; but I agree that would work out just fine.

Reply:
Does `DEFAULT_stage` and `DEFAULT_modifiers` sound okay? Any other suggestions?